### PR TITLE
goto Implementations with local alias types

### PR DESCRIPTION
**Pull request by eilite**
This PR addresses https://github.com/scalameta/metals/issues/1464

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala [hunk 1]
```diff
@@ -338,17 +338,29 @@ final class ImplementationProvider(
...
+            // local symbols inheritance should only be picked up in the same file
+            (loc.symbol.isLocal && loc.file != currentPath)
+          )
       }
       directImplementations ++ directImplementations
         .flatMap { loc =>
-          val allPossible = loop(loc.symbol, loc.file)
+          val allPossible = loop(
+            loc.symbol,
+            loc.file.orElse(Some(file).filter(_ => isTypeAlias(loc.symbol)))
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala:362]: It seems we are missing the file due to [here](https://github.com/scalameta/metals/blob/39fa3cc3acbed1b7e252747e520ce42ad542260f/metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala#L490)

I think this was done in order not to show types as implementations. We could add the file the same way we add it in the method above and move checking for type alias in [here](https://github.com/scalameta/metals/blob/39fa3cc3acbed1b7e252747e520ce42ad542260f/metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala#L323)

We should use the `implDocument` to search for the `SymbolInformation` and check if it is a type alias.

File: metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala [hunk 2]
```diff
@@ -457,6 +468,12 @@ object ImplementationProvider {
       .find(sym => sym.symbol == symbol)
   }
 
+  def isTypeAlias(
+      symbol: String,
+      findSymbol: String => Option[SymbolInformation]
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala:473]: We could use the already existing `implDocument` and have it passed here. Then:
`ImplementationProvider.findSymbol(symbol, implDocument).exists(_.kind == SymbolInformation.Kind.TYPE)`

File: metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala [hunk 3]
```diff
@@ -320,7 +320,18 @@ final class ImplementationProvider(
         classContext,
         implReal
       )
-      implOccurrence <- findDefOccurrence(implDocument, implSymbol, source)
+      symbolSearch = defaultSymbolSearch(
+        implPath,
+        implDocument
+      )
+      filteredImplSymbol <- Some(implSymbol).filterNot(
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala:327]: We could just do `if ! isTypeAlias(implSymbol, implDocument) `

Comment (eilite) [reply metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala:327]: Oh yeah of course !

**Comment by tgodzik** [comment]
Just noticed, that one symbol was unused, so I removed it and I think we are ready to merge.

**Pull request merged**